troposphere/__init__.py

```diff
@@ -38,9 +38,9 @@ def __init__(self, title, template=None, **kwargs):
                            'Metadata', 'UpdatePolicy',
                            'Condition', 'CreationPolicy']
 
-        # unset/None is also legal
-        if title and not valid_names.match(title):
-            raise ValueError('Name "%s" not alphanumeric' % title)
+        # try to validate the title if its there
+        if self.title is not None:
```

Comment:
`if self.title` is more pythonic, and should do what you want here.  Otherwise I think this is good to go.